### PR TITLE
Pull image from authenticated registry

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

Pull image from authenticated registry

## Test coverage
- No tests

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23297

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: shared/utils/exec.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// RedactedWords is a collection of word redacted by logs.
+var RedactedWords = []string{}
```
Comment (cbosdo) [inline shared/utils/exec.go:17]: Why do we want this dynamic? Couldn't it just be initialized with all the words to redact?

File: shared/utils/exec.go [hunk 2]
```diff
@@ -49,9 +52,23 @@ func RunCmdStdMapping(command string, args ...string) error {
...
 func RunCmdOutput(logLevel zerolog.Level, command string, args ...string) ([]byte, error) {
-	log.Debug().Msgf("Running: %s %s", command, strings.Join(args, " "))
+	log.Debug().Msgf("Running: %s %s", command, strings.Join(filterRedactedArgs(args...), " "))
 
 	output, err := exec.Command(command, args...).Output()
 	log.Trace().Msgf("Command output: %s, error: %s", output, err)
 	return output, err
 }
+
+func filterRedactedArgs(args ...string) []string {
```
Comment (cbosdo) [inline shared/utils/exec.go:62]: It would be nice to have a unit test for this function

File: mgradm/shared/utils/exec.go [hunk 3]
```diff
@@ -180,3 +201,92 @@ func RunningImage(cnx *shared.Connection, containerName string) (string, error)
...
+		return map[string]string{}, fmt.Errorf("failed to run inspect script in host system: %s", err)
+	}
+
+	inspectResult, err := ReadInspectData(scriptDir, "host_")
+	if err != nil {
+		return map[string]string{}, fmt.Errorf("cannot inspect host data. %s", err)
+	}
+
+	for key, value := range inspectResult {
+		if strings.Contains(key, "password") {
```
Comment (rjmateus) [inline mgradm/shared/utils/exec.go:258]: Is it filtering the key by password? But don't we need to read also the user name?

Comment (mbussolotto) [reply mgradm/shared/utils/exec.go:258]: yes, we're also reading username. This function is just to redact password from logs

**Comment by mbussolotto** [comment]
@cbosdo I remove the filter logic: I think it would be simpler just to change the loglevel (only if the args are present). I also fixed `RunCmdOutput`: I think it was not working as expected, because the loglevel argument was ignored.

File: mgradm/shared/utils/exec.go (hunk 3 above)
Comment (cbosdo) [reply mgradm/shared/utils/exec.go]: redacting user may be useful too

File: mgradm/cmd/inspect/podman.go [hunk 4]
```diff
@@ -62,27 +61,39 @@ func InspectPodman(serverImage string, pullPolicy string) (map[string]string, er
...
+	inspectedHostValues, err := adm_utils.InspectHost()
+	if err != nil {
+		return map[string]string{}, fmt.Errorf("cannot inspect host values: %s", err)
+	}
+
+	pullArgs := []string{}
+	_, scc_user_exist := inspectedHostValues["host_scc_username"]
+	_, scc_user_password := inspectedHostValues["host_scc_password"]
+	if scc_user_exist && scc_user_password {
+		pullArgs = append(pullArgs, "--creds", inspectedHostValues["host_scc_username"]+":"+inspectedHostValues["host_scc_password"])
```
Comment (cbosdo) [inline mgradm/cmd/inspect/podman.go:73]: creds would need to be redacted too

File: shared/podman/images.go [hunk 5]
```diff
@@ -48,10 +48,18 @@ func checkImage(image string, pullPolicy string) (bool, error) {
...
-func pullImage(image string) error {
+func pullImage(image string, args ...string) error {
 	log.Info().Msgf("Running podman pull %s", image)
+	podmanImageArgs := []string{"pull", image}
+	podmanArgs := append(podmanImageArgs, args...)
 
-	return utils.RunCmdStdMapping("podman", "pull", image)
+	loglevel := zerolog.DebugLevel
+	if len(args) > 0 {
+		loglevel = zerolog.Disabled
```
Comment (cbosdo) [inline shared/podman/images.go:58]: Ok, so if there are credentials we don't show anything? why not... we can still improve later if that turns out painful for debugging

**Review by cbosdo — APPROVED** [review]

**Pull request merged**